src/asset/material/builder.rs

```diff
@@ -367,6 +367,66 @@ fn build_water_material(
     builder.build()
 }
 
+fn build_modulate_material(
+    context: &mut Context<BlenderAssetHandler>,
+    vmt: &VmtHelper,
+    settings: Settings,
+) -> BuiltMaterialData {
+    let shader = vmt.shader().shader.as_uncased_str();
+    let is_mod2x = shader == "decalmodulate".as_uncased()
+        || (shader == "modulate".as_uncased() && vmt.extract_param_or_default::<bool>("$mod2x"));
+
+    let mut builder = if is_mod2x {
+        // For mod2x materials, use transparent shader for proper blending
+        MaterialBuilder::new(&shaders::TRANSPARENT)
+    } else {
+        // For regular modulate, use principled shader with multiply blend
+        MaterialBuilder::new(&shaders::PRINCIPLED)
+    };
+
+    // Handle the base texture
+    let color_space = if is_mod2x {
+        // Use non-color space to preserve 50% gray values for mod2x
+        ColorSpace::NonColor
+    } else {
+        // Use sRGB for regular modulate
+        ColorSpace::Srgb
+    };
+
+    if builder.handle_texture(
+        context,
+        vmt,
+        "$basetexture",
+        Some("$basetexturetransform"),
+        color_space,
+        settings.texture_interpolation,
+    ) {
+        if is_mod2x {
+            // Apply mod2x operation: multiply color by 2 to convert 50% gray to white
+            let output = builder.output("Color", "$basetexture", "color");
+            output
+                .push(&groups::MOD2X)
+                .link_input(&groups::MOD2X, "color");
+        } else {
+            // For regular modulate, connect texture directly to base color
+            builder.output("Base Color", "$basetexture", "color");
+        }
+    }
+
+    // Handle alpha for transparency
+    if builder.has_input("$basetexture") {
```

Comment:
According to https://developer.valvesoftware.com/wiki/Modulate alpha parameter is not supported, just remove it